TempleEight/docs/arch/system.md

```diff
@@ -3,3 +3,189 @@ id: system
 title: System Architecture
 sidebar_label: System Architecture
 ---
+
+import useBaseUrl from '@docusaurus/useBaseUrl';
+
+On top of generating service-level application code, Temple produces orchestration configuration files to enable executing the services together as a whole system.
+This guide will take you through how a Temple generated application is designed on the super-service level.
+We'll look at how the system is organised and how it changes with more advanced features.
+
+The details of how the application code in each service is architected can be seen in the [Service Architecture](service) reference.
+
+## Minimal Example
+
+We'll start with the example Templefile from the [Getting Started](../getting-started) Guide.
+
+```templefile
+ExampleProject: project {
+  #language(go);
+  #database(postgres);
+  #provider(dockerCompose);
+}
+
+ExampleService: service {
+  foo: string;
+  bar: int;
+}
+```
+
+Also from the Getting Started Guide is this diagram outlining the system architecture:
+
+<img alt="Tutorial System Architecture" src={useBaseUrl('img/tutorial-architecture.png')} />
+
+It shows a high level description of the system, detailing how the [Kong API Gateway](https://konghq.com/kong/) sits between the user and the backend services, and how the service communicates with its database.
+
+Let's delve a little bit deeper into this architecture and see some finer details:
+
+<img alt="System Architecture Deeper" src={useBaseUrl('img/system-architecture-basic.png')} />
+
+We can see that all components of our application are executed inside of [Docker](https://www.docker.com/) containers, including Kong.
+This allows us to not have to worry about installing it locally, giving this job to our orchestration tool. 
+Containers allow us to encapsulate our application components and their dependencies, allowing them to be easily ran anywhere, among many other benefits.
+
+We can also see that Kong uses it's own database to store information about which services exist and how to route requests to them.
+Like the rest of the Kong infrastructure, this database is orchestrated automatically by Temple.
+A [`PostgreSQL`](https://www.postgresql.org/) instance sits alongside Kong on a private network, so that only the Kong instance can communicate with it.
```

Comment:
Might be worth linking Kong docs at the end, "for more information on Kong...", especially since Docker is linked above

Reply:
There's a link at the beginning of this paragraph :)